android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java

```diff
@@ -49,12 +49,12 @@ public static String getPackageID(File path) {
 
     if(filename.lastIndexOf('.') != -1) {
       // Android's temp downloads attach a suffix to the extension; .kmp isn't the end of the filename.
-      if(filename.lastIndexOf(".kmp") == -1) {
+      if(!filename.toLowerCase().contains(".kmp")) {
```

Comment:
Shouldn't we be using `FileUtils` here too? `.contains(".kmp")` will return true for a name like "foo.kmp.blah.js"...